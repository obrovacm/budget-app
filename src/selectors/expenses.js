import moment from "moment";

// Get visible expenses
export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter(expense => {
      const createdMoment = moment(expense.createdAt);

      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdMoment, "day")
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdMoment, "day")
        : true;

      const textMatchNote =
        typeof text !== "string" ||
        expense.note.toLowerCase().includes(text.toLowerCase());
      const textMatchDescription =
        typeof text !== "string" ||
        expense.description.toLowerCase().includes(text.toLowerCase());

      return (
        startDateMatch &&
        endDateMatch &&
        (textMatchNote || textMatchDescription)
      );
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createdAt < b.createdAt ? 1 : -1;
      }
      // else if (sortBy === "amount") {
      //   return a.amount < b.amount ? 1 : -1;
      // }

      // this way there's no 'default' return warning
      return a.amount < b.amount ? 1 : -1;
    });
};
