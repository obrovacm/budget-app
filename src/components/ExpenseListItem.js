import React from "react";
import { connect } from "react-redux";
import { removeExpense } from "../actions/expenses";

const ExpenseListItem = ({
  id,
  description,
  note,
  amount,
  createdAt,
  dispatch
}) => (
  <div className="expense-list-item">
    <hr />
    <h3>{description}</h3>
    <h5>${amount}</h5>
    <h5>date:{createdAt}</h5>
    <p>{note}</p>
    <button
      onClick={() => {
        dispatch(removeExpense({ id }));
      }}
    >
      remove
    </button>
  </div>
);

export default connect()(ExpenseListItem);
