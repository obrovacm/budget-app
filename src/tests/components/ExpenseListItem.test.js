import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render ExpenseListItem correctly", () => {
  const wrapper = shallow(
    <ExpenseListItem expenses={expenses[0]} key={expenses[0].id} />
  );
  expect(wrapper).toMatchSnapshot();
});
