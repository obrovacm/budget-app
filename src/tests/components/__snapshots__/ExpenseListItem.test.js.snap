// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<div
  className="expense-list-item"
>
  <hr />
  <Link
    to="/edit/undefined"
  >
    <h3 />
  </Link>
  <h5>
    $
     - 
    21.01.2020.
  </h5>
  <p />
</div>
`;
